agent/hcp/telemetry/otel_exporter.go

```diff
@@ -15,15 +15,15 @@ import (
 // The exporter is used by a OTEL Metrics SDK PeriodicReader to export aggregated metrics.
 // This allows us to use a custom client - HCP authenticated MetricsClient.
 type OTELExporter struct {
-	client hcpclient.MetricsClient
-	url    url.URL
+	client   hcpclient.MetricsClient
+	endpoint *url.URL
```

Comment:
Updated to use a pointer `*url.URL` and the same url naming was a conflict with the library 😓 so switched to endpoint.